fix(property-filter): use a subscription to update the page title

mariusandra opened the pull request:
## Problem

Do not ask me how, but this fixes the following.

## Changes

With this fix:

![2022-04-11 17 18 04](https://user-images.githubusercontent.com/53387/162771723-de531d1c-d9e1-48f4-be0d-f2b45e2a9c3c.gif)

On master:

![2022-04-11 17 18 38](https://user-images.githubusercontent.com/53387/162771805-79e025fa-6581-448f-82e1-eabbb9383fab.gif)

## How did you test this code?

Eyes closed.

mariusandra committed:
fix(property-filter): use a subscription to update the page title

EDsCODE commented on frontend/src/layout/navigation/Breadcrumbs/breadcrumbsLogic.tsx:
```diff
@@ -146,5 +146,25 @@ export const breadcrumbsLogic = kea<breadcrumbsLogicType>({
+
+    subscriptions: ({}: breadcrumbsLogicType) => ({
+        titleBreadcrumbs: (titleBreadcrumbs: Breadcrumb[]) => {
+            if (typeof document !== 'undefined') {
```
Maybe this solved it? Otherwise it seems like the logic would be the same

EDsCODE reviewed (approved):
I'm also not 100% certain why moving the logic here solves it but it fixes the bug and keeps more logic in kea so I'm ok with this

mariusandra committed:
store document title as a string value

mariusandra committed:
mount breadcrumbs with app

mariusandra commented:
Your hunch might be right, though it's still weird how this happens. It's possible adding another selector and using `[stringTitle]` for `useEffect` might have solved things. However I was too committed, and refactored this further... and added a test.

Unfortunately the `sceneBreadcrumbs` selector is recalculated way too often. Somebody also [requested a fix](https://github.com/keajs/kea/issues/143) for this. For us, for now, just having `documentTitle` a string seems to work.

mariusandra committed:
add test

Twixes reviewed (approved):
Looks proper 👌
